quodlibet/quodlibet/formats/_audio.py

```diff
@@ -532,26 +533,35 @@ def list(self, key):
             v = self.get(key)
             return [] if v is None else v.split("\n")
 
-    def list_separate(self, key, connector=" - "):
-        """Similar to list, but will return a list of all combinations
-        for tied tags instead of one comma separated string.
-
-        In case of tied tags the result will be unicode, otherwise
-        it returns the same as list()
+    def list_sort(self, key):
+        """Like list but return display,sort pairs when appropriate
+        and work on all tags
         """
+        display = decode_value(key, self.__call__(key))
+        display = display.split("\n") if display else []
+        sort = []
+        if key in TAG_TO_SORT:
+            sort = decode_value(TAG_TO_SORT[key],
+                                self.__call__(TAG_TO_SORT[key]))
+            # it would be better to use something that doesn't fall back
+            # to the key itself, but what?
+            sort = sort.split("\n") if sort else []
+        if len(display) > 0 and len(sort) > 0 and display != sort:
+            return [((d if d != "" else s, s) if s != "" else d)
+                    for d, s in izip_longest(display, sort, fillvalue="")]
+        else:
```

Comment:
I think it would be nicer to just always return a list of tuples even if there are no sort values. Also throw out sort values if there are too many.